core/remote_state_verifier.go

```diff
@@ -69,14 +71,26 @@ func NewVerifyManager(blockchain *BlockChain, peers verifyPeers, allowInsecure b
 		if oldBlock == nil {
 			return nil, fmt.Errorf("block is nil, number: %d", number)
 		}
-		blockHash := oldBlock.Hash()
-		_, err := blockchain.GenerateDiffLayer(blockHash)
-		if err != nil {
-			return nil, err
+
+		// rewind to last non verified block
+		blockchain.SetHead(oldBlock.NumberU64())
```

Comment:
It seems we don't need a loop here, just check the number, if number > maxForkHeight, rewind to number-maxForkHeight is OK, if not rewind to a genesis block?

Reply:
fixed